### PR TITLE
ConseilJS 5.0.2-beta.3

### PR DESCRIPTION
**Pull request by anonymoussprocket**
## Changes

- updated Signer interface for signing text hashes
- updated example for SoftSigner usage
- added [TezosConseilClient.getBigMapValueForKey](https://github.com/Cryptonomic/ConseilJS/blob/release/502/src/reporting/tezos/TezosConseilClient.ts#L312)

### CODE REVIEW

File: src/reporting/tezos/TezosConseilClient.ts [hunk 1]
```diff
@@ -296,7 +296,43 @@ export namespace TezosConseilClient {
...
 
-        return { contract: accountID, maps };
+        return { contract, maps };
+    }
+
+    /**
+     * Returns a value for a given key. The big map id is either provided as a paramter or the smallest of the possibly multiple big maps associated with the given contract address is used. Note that sometimes key values must be wrapped in quotes for key types that are non-numeric and not byte type.
+     * 
+     * @param serverInfo Conseil server connection definition.
+     * @param key Key to query for.
```
Comment (vishakh) [inline src/reporting/tezos/TezosConseilClient.ts:306]: We should specify whether this is encoded or not.

File: src/reporting/tezos/TezosConseilClient.ts [hunk 2]
```diff
@@ -296,7 +296,43 @@ export namespace TezosConseilClient {
...
+
+            const ownerResult = await getTezosEntityData(serverInfo, serverInfo.network, 'originated_account_maps', ownerQuery);
+
+            if (ownerResult.length < 1) { throw new Error(`Could not find any maps for ${contract}`); }
+            mapIndex = ownerResult[0];
+        }
+
+        let contentQuery = ConseilQueryBuilder.setLimit(ConseilQueryBuilder.blankQuery(), 1);
+        contentQuery = ConseilQueryBuilder.addFields(contentQuery, 'value');
+        contentQuery = ConseilQueryBuilder.addPredicate(contentQuery, 'key', ConseilOperator.EQ, [key], false);
```
Comment (vishakh) [inline src/reporting/tezos/TezosConseilClient.ts:329]: Is it not possible to automatically encode the key? I forget the details.

File: src/reporting/tezos/TezosConseilClient.ts (hunk 1 above)
Comment (anonymoussprocket) [reply src/reporting/tezos/TezosConseilClient.ts:308]: docs were updated

File: src/reporting/tezos/TezosConseilClient.ts (hunk 2 above)
Comment (anonymoussprocket) [reply src/reporting/tezos/TezosConseilClient.ts:331]: not at this time, contract parser will need updates.

**Pull request merged**